daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-bulk.md

```diff
@@ -0,0 +1,285 @@
+      // Create a list of messages to publish
+      List<String> messages = new ArrayList<>();
+      for (int i = 0; i < 10; i++) {
+        String message = String.format("This is message #%d", i);
+        messages.add(message);
+      }
+
+      // Publish list of messages using the bulk publish API
+      BulkPublishResponse<String> res = client.publishEvents(PUBSUB_NAME, TOPIC_NAME, "text/plain", messages).block();
+    }
+  }
+}
+```
+
+{{% /codetab %}}
+
+{{% codetab %}}
+
+```typescript
+
+import { DaprClient } from "@dapr/dapr";
+
+const pubSubName = "my-pubsub-name";
+const topic = "topic-a";
+
+async function start() {
+    const client = new DaprClient();
+
+    // Publish multiple messages to a topic.
+    await client.pubsub.publishBulk(pubSubName, topic, ["message 1", "message 2", "message 3"]);
+
+    // Publish multiple messages to a topic with explicit bulk publish messages.
+    const bulkPublishMessages = [
+    {
+      entryID: "entry-1",
+      contentType: "application/json",
+      event: { hello: "foo message 1" },
+    },
+    {
+      entryID: "entry-2",
+      contentType: "application/cloudevents+json",
+      event: { 
+        specversion: "1.0",
+        source: "/some/source",
+        type: "example",
+        id: "1234", 
+        data: "foo message 2", 
+        datacontenttype: "text/plain" 
+      },
+    },
+    {
+      entryID: "entry-3",
+      contentType: "text/plain",
+      event: "foo message 3",
+    },
+  ];
+  await client.pubsub.publishBulk(pubSubName, topic, bulkPublishMessages);
+}
+
+start().catch((e) => {
+  console.error(e);
+  process.exit(1);
+});
+```
+
+{{% /codetab %}}
+
+{{% codetab %}}
+
+```bash
+curl -X POST http://localhost:3500/v1.0-alpha1/publish/bulk/my-pubsub-name/topic-a \
+  -H 'Content-Type: application/json' \
+  -d '[
+        {
+            "entryId": "ae6bf7c6-4af2-11ed-b878-0242ac120002",
+            "event":  "first",
+            "contentType": "text/plain"
+        },
+        {
+            "entryId": "b1f40bd6-4af2-11ed-b878-0242ac120002",
+            "event":  {
+                "message": "second"   
+            },
+            "contentType": "application/json"
+        },
+      ]'
+```
+
+{{% /codetab %}}
+
+{{% codetab %}}
+
+```powershell
+Invoke-RestMethod -Method Post -ContentType 'application/json' -Uri 'http://localhost:3500/v1.0-alpha1/publish/bulk/my-pubsub-name/topic-a' `
+-Body '[
+        {
+            "entryId": "ae6bf7c6-4af2-11ed-b878-0242ac120002",
+            "event":  "first",
+            "contentType": "text/plain"
+        },
+        {
+            "entryId": "b1f40bd6-4af2-11ed-b878-0242ac120002",
+            "event":  {
+                "message": "second"   
+            },
+            "contentType": "application/json"
+        },
+      ]'
+```
+
+{{% /codetab %}}
+
+{{< /tabs >}}
+
+## Subscribing messages in bulk
+
+The bulk subscribe API allows you to subscribe multiple messages from a topic in a single request. 
```

Comment:
```suggestion
The bulk subscribe API allows you to receive multiple messages from a topic in a single request. 
```

Reply:
There are going to be a number of changes to the original PRs, so need to get through those first which will affect this one.